### PR TITLE
Edit sampling

### PR DESCRIPTION
**Pull request by Dpananos**
As per #3037 

* Renamed `sample_ppc` to `sample_posterior_predictive`. `sample_ppc` as an alias for `sample_posterior_predictive`and included deprecation warning.

* Ditto for `sample_ppc_w`.

* Made appropriate changes to tests, notebooks, and docs (I think.  I used `git grep -l 'sample_ppc' | xargs sed -i '' -e 's/sample_ppc/sample_posterior_predictive/g' `)

### CODE REVIEW

File: .bash_history [hunk 1]
```diff
@@ -0,0 +1,17 @@
+cd ~/pymc3
```
Comment (twiecki) [inline .bash_history]: this shouldn't be here.

File: pymc3/sampling.py [hunk 2]
```diff
@@ -27,7 +27,14 @@
 import sys
 sys.setrecursionlimit(10000)
 
-__all__ = ['sample', 'iter_sample', 'sample_ppc', 'sample_ppc_w', 'init_nuts', 'sample_prior_predictive']
+__all__ = [ 'sample',
+			'iter_sample',
```
Comment (twiecki) [inline pymc3/sampling.py]: bad indent

**Comment by twiecki** [comment]
Couple of wrong files committed.

File: pymc3/sampling.py (hunk 2 above)
Comment (Dpananos) [reply pymc3/sampling.py]: That indent doesn't look that way on my text editor.  I could just leave them all on the same line, but that would violate pep8.

**Comment by Dpananos** [comment]
re: Intents, this is what shows when I commit the changes

image

No clue why the indents look as they do in the PR.  I'll keep trying things,

**Comment by Dpananos** [comment]
OK, I think that should be good.

**Comment by eigenfoo** [comment]
@Dpananos about your indents, it looks like you used tabs instead of spaces. Your editor is probably configured to make tabs _look_ like four spaces to you, but really you entered a tab character, which explains why it looks different once you commit it. Per pep8, four spaces is preferred over tabs, so if you do a lot of Python coding, I'd suggest you reconfigure your editor 😃

(Sorry to swoop in, but I had this exact same problem a while back and it frustrated me to no end)

**Comment by Dpananos** [comment]
Ah, that’s brilliant!  Spaces it is then, I’ll adjust the PR

**Comment by eigenfoo** [comment]
No problem! I'm actually mildly surprised that `pylint` didn't catch that.

File: .coverage [hunk 3]
```diff
@@ -0,0 +1 @@
...uning/__init__.py":[],"/home/jovyan/pymc3/external/__init__.py":[],"/home/jovyan/pymc3/glm/linear.py":[],"/home/jovyan/pymc3/glm/families.py":[],"/home/jovyan/pymc3/glm/__init__.py":[],"/home/jovyan/pymc3/glm/utils.py":[],"/home/jovyan/pymc3/distributions/transforms.py":[],"/home/jovyan/pymc3/distributions/mixture.py":[],"/home/jovyan/pymc3/distributions/dist_math.py":[],"/home/jovyan/pymc3/distributions/discrete.py":[],"/home/jovyan/pymc3/distributions/distribution.py":[],"/home/jovyan/pymc3/distributions/multivariate.py":[],"/home/jovyan/pymc3/distributions/continuous.py":[],"/home/jovyan/pymc3/distributions/special.py":[],"/home/jovyan/pymc3/distributions/bound.py":[],"/home/jovyan/pymc3/plots/kdeplot.py":[],"/home/jovyan/pymc3/plots/posteriorplot.py":[],"/home/jovyan/pymc3/plots/autocorrplot.py":[],"/home/jovyan/pymc3/plots/pairplot.py":[],"/home/jovyan/pymc3/plots/densityplot.py":[],"/home/jovyan/pymc3/plots/__init__.py":[],"/home/jovyan/pymc3/plots/compareplot.py":[],"/home/jovyan/pymc3/plots/utils.py":[],"/home/jovyan/pymc3/plots/forestplot.py":[],"/home/jovyan/pymc3/plots/energyplot.py":[],"/home/jovyan/pymc3/plots/artists.py":[],"/home/jovyan/pymc3/plots/traceplot.py":[]}}
```
Comment (twiecki) [inline .coverage]: remove

File: .local/share/jupyter/runtime/nbserver-6.json [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (twiecki) [inline .local/share/jupyter/runtime/nbserver-6.json]: remove

File: .local/share/jupyter/runtime/notebook_cookie_secret [hunk 5]
```diff
@@ -0,0 +1 @@
+UuY3/XZFZ/GeLSUii4IXE4ffamC+usYZkh41lyELfzg=
```
Comment (twiecki) [inline .local/share/jupyter/runtime/notebook_cookie_secret]: remove

File: pymc3/sampling.py [hunk 6]
```diff
@@ -27,7 +27,9 @@
 import sys
 sys.setrecursionlimit(10000)
 
-__all__ = ['sample', 'iter_sample', 'sample_ppc', 'sample_ppc_w', 'init_nuts', 'sample_prior_predictive']
+__all__ = ['sample', 'iter_sample', 'sample_posterior_predictive',
+            'sample_posterior_predictive_w', 'init_nuts',
```
Comment (twiecki) [inline pymc3/sampling.py]: one less space :)

File: RELEASE-NOTES.md [hunk 7]
```diff
@@ -87,10 +87,10 @@
 
 - Fixed `compareplot` to use `loo` output.
 - Improved `posteriorplot` to scale fonts
-- `sample_ppc_w` now broadcasts
+- `sample_posterior_predictive_w` now broadcasts
```
Comment (twiecki) [inline RELEASE-NOTES.md]: Actually we probably want to keep those intact, for historical reasons.

But we definitely want a new item in here that says that it's renamed for the next release.

Comment (Dpananos) [reply RELEASE-NOTES.md]: That must have been switched when I did the find replace. I’ll edit that today.

Comment (Dpananos) [reply RELEASE-NOTES.md]: @twiecki  How would you like me to add the name changes in the `RELEASE-NOTES.md`?  Should I start a new header?

Comment (twiecki) [reply RELEASE-NOTES.md]: @Dpananos At the top (https://github.com/pymc-devs/pymc3/pull/3123/files/3f785dc4dd5dc3d2f6be3ea5030bde709663b38d#diff-89f3cba2993eebd391f32c27f3772d7bR2) add a new item for PyMC3 3.6 and put "* Renamed `sample_ppc()` and `sample_ppc_w()` to `sample_posterior_predictive()` and sample_posterior_predictive_w()`, respectively."

File: RELEASE-NOTES.md [hunk 8]
```diff
@@ -1,8 +1,13 @@
 # Release Notes
 
 
+## PyMC3 3.6
+
 ## PyMC 3.5 (July 21 2018)
 
+- Renamed `sample_ppc()` and `sample_ppc_w()` to `sample_posterior_predictive()` and `sample_posterior_predictive_w()`, respectively.
```
Comment (twiecki) [inline RELEASE-NOTES.md]: This should be under the new 3.6. heading.

**Comment by Dpananos** [comment]
Thanks for your patience.

**Comment by twiecki** [comment]
LGTM, will merge when tests pass.

**Comment by twiecki** [comment]
Thanks @Dpananos!

**Comment by Dpananos** [comment]
Thanks for your patience and help!

Demetri Pananos
Ph.D Candidate
Department of Epidemiology and Biostatistics
Schulich School of Medicine and Dentistry
Western University

**Comment by springcoil** [comment]
Just saw this @Dpananos Thanks for your contribution :)

**Pull request merged**